src/Simplify/RangeDict.elm

```diff
@@ -1,10 +1,11 @@
-module Simplify.RangeDict exposing (RangeDict, empty, get, insert, member)
+module Simplify.RangeDict exposing (RangeDict, any, diff, empty, fromList, get, insert, map, mapFromList, member, singleton, union)
 
 import Dict exposing (Dict)
 import Elm.Syntax.Range exposing (Range)
 
 
 type alias RangeDict v =
+    -- TODO: make opaque to disallow type confusion with other String dicts
     Dict String v
```

Comment:
What's your opinion? The idea of `RangeDict` in general seems to match with [`miniBill/elm-generic-dict`](https://dark.elm.dmy.fr/packages/miniBill/elm-generic-dict/latest/)